crates/astria-sequencer/src/ibc/ics20_withdrawal.rs

```diff
@@ -69,47 +69,53 @@ fn withdrawal_to_unchecked_ibc_packet(
 /// Establishes the withdrawal target.
 ///
 /// The function returns the following addresses under the following conditions:
-/// 1. `from` if `action.bridge_address` is unset and `from` is *not* a bridge account;
-/// 2. `from` if `action.bridge_address` is unset and `from` is a bridge account and `from` is its
-///    stored withdrawer address.
-/// 3. `action.bridge_address` if `action.bridge_address` is set and a bridge account and `from` is
-///    its stored withdrawer address.
+/// 1. `action.bridge_address` if `action.bridge_address` is set and `from` is its stored withdrawer
+///    address.
+/// 2. `from` if `action.bridge_address` is unset and `from` is *not* a bridge account.
+///
+/// Errors if:
+/// 1. Errors reading from DB
+/// 2. `action.bridge_address` is set, but `from` is not the withdrawer address.
+/// 3. `action.bridge_address` is unset, but `from` is a bridge account.
 async fn establish_withdrawal_target<S: StateRead>(
     action: &action::Ics20Withdrawal,
     state: &S,
     from: [u8; 20],
 ) -> Result<[u8; 20]> {
-    if action.bridge_address.is_none()
-        && !state
-            .is_a_bridge_account(from)
+    // If the bridge address is set, the withdrawer on that address must match
+    // the from address.
+    if let Some(bridge_address) = action.bridge_address {
+        let Some(withdrawer) = state
+            .get_bridge_account_withdrawer_address(bridge_address)
             .await
-            .context("failed to get bridge account rollup id")?
-    {
-        return Ok(from);
-    }
+            .context("failed to get bridge withdrawer")?
+        else {
+            bail!("bridge address must have a withdrawer address set");
+        };
 
-    // if `action.bridge_address` is set, but it's not a valid bridge account,
-    // the `get_bridge_account_withdrawer_address` step will fail.
-    let bridge_address = action.bridge_address.map_or(from, Address::bytes);
+        ensure!(
+            withdrawer == from.address_bytes(),
+            "sender does not match bridge withdrawer address; unauthorized"
+        );
 
-    let Some(withdrawer) = state
-        .get_bridge_account_withdrawer_address(bridge_address)
-        .await
-        .context("failed to get bridge withdrawer")?
-    else {
-        bail!("bridge address must have a withdrawer address set");
-    };
+        return Ok(bridge_address.address_bytes());
+    }
 
-    ensure!(
-        withdrawer == from.address_bytes(),
-        "sender does not match bridge withdrawer address; unauthorized"
-    );
+    // If the bridge address is not set, the sender must not be a bridge account.
+    if state
+        .is_a_bridge_account(from)
+        .await
+        .context("failed to get bridge account rollup id")?
```

Comment:
Update this error message (here you don't know what the called method is actually doing):
```suggestion
        .context("failed to establish whether the sender is a bridge account")?
```